### PR TITLE
Add app icon and splash screen in FabricExample

### PR DESCRIPTION
**Pull request by jfedak**
## Summary

Previously, there was no app icon nor splash screen in FabricExample app. This PR adds those things both on iOS and Android. Due to the usage of different icon, users can distinguish ReanimatedExample app from FabricExample app.

iOS:
New app icon was placed in `Images.xcassets` directory. Splash screen uses the configuration from `LaunchScreen.storyboard`.

https://github.com/software-mansion/react-native-reanimated/assets/56112901/b5297452-5efc-4d50-a154-4a6318519fb0

Android:
Old icon was replaced with new one inside `res/mipmap`. The splash screen was implemented with 
[androidx.core:core-splashscreen](https://developer.android.com/jetpack/androidx/releases/core).

https://github.com/software-mansion/react-native-reanimated/assets/56112901/21816d81-ffbc-4500-b398-37377b06d2d2

https://github.com/software-mansion/react-native-reanimated/assets/56112901/44e8599d-06c0-460f-ada0-bfd523c61bd0

### CODE REVIEW

File: FabricExample/android/app/src/main/res/mipmap-anydpi-v26/ic_launcher.xml [hunk 1]
```diff
@@ -0,0 +1,5 @@
+<?xml version="1.0" encoding="utf-8"?>
+<adaptive-icon xmlns:android="http://schemas.android.com/apk/res/android">
+    <background android:drawable="@color/ic_launcher_background"/>
+    <foreground android:drawable="@mipmap/ic_launcher_foreground"/>
+</adaptive-icon>
```
Comment (tjzel) [inline FabricExample/android/app/src/main/res/mipmap-anydpi-v26/ic_launcher.xml:5]: Is there any particular reason we don't have trailing new lines at the end of the file? I checked and it's the same in Example - just wondering 🤔

**Review by tjzel — APPROVED** [review]
Everything looks good 🥳

**Review by tomekzaw — APPROVED** [review]
Congratulations on the first PR! 🥳

File: FabricExample/android/app/src/main/res/mipmap-anydpi-v26/ic_launcher.xml (hunk 1 above)
Comment (jfedak) [reply FabricExample/android/app/src/main/res/mipmap-anydpi-v26/ic_launcher.xml:5]: Those files are autogenerated by Android Studio when an image asset is created. The same thing happens when you create new resource file, i.e. there is no trailing new line.

**Pull request merged**